src/shelter/use-app-lock.hook.ts

```diff
@@ -30,12 +37,26 @@ export const useAppLock = () => {
     const subscriptions = [
       Shelter.isLocked$.subscribe(value => setIsLocked(value)),
       unlock$
-        .pipe(switchMap(password => Shelter.unlockApp$(password)))
-        .subscribe(success => !success && showErrorToast({ description: 'Wrong password, please, try again' }))
+        .pipe(
+          delay(passwordDelay),
+          switchMap(password => Shelter.unlockApp$(password))
+        )
+        .subscribe(success => {
+          if (success) {
+            dispatch(enterPassword.success());
+
+            return true;
+          } else {
+            dispatch(enterPassword.fail(''));
```

Comment:
you could omit error message `export const enterPassword = createActions<void, void, void>('security/ENTER_PASSWORD');`